x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/Transform.java

```diff
@@ -137,6 +139,23 @@
         Setting.Property.Dynamic
     );
 
+    /**
+     * Node attributes for transform, automatically created and retrievable via cluster state.
+     * These attributes should never be set directly, use the node setting counter parts instead.
+     */
+    public static final String TRANSFORM_ENABLED_NODE_ATTR = "transform.node";
+    public static final String TRANSFORM_REMOTE_ENABLED_NODE_ATTR = "transform.remote_connect";
+
+    /**
+     * Setting whether transform (the coordinator task) can run on this node and REST API's are available,
+     * respects xpack.transform.enabled (for the whole plugin) as fallback
+     */
+    public static final Setting<Boolean> TRANSFORM_ENABLED_NODE = Setting.boolSetting(
+        "node.transform",
+        XPackSettings.TRANSFORM_ENABLED,
```

Comment:
I realised that in my previous review where I suggested _not_ taking `node.data` into account in the assignment that this could cause a surprise to users who don't adjust their `elasticsearch.yml` after upgrading to 7.7, because it would mean transforms would start running on all types of nodes instead of just data nodes.

So an additional change I would suggest to prevent this is that the default value of `node.transform` be `xpack.transform.enabled && node.data`.  In other words, if `node.transform` is not explicitly specified then it is `false` if either `xpack.transform.enabled` is `false` or `node.data` is `false`, and `true` otherwise.

This complex default value can be implemented using the `Setting` factory method that takes a `Function` for the default value: https://github.com/elastic/elasticsearch/blob/012746dd816a5ab6555b6bf3b7856d34974f50a5/server/src/main/java/org/elasticsearch/common/settings/Setting.java#L1264